tx.go

```diff
@@ -322,32 +330,40 @@ func (tx *Tx) Clone() *Tx {
 // NodeJSON returns a wrapped *bt.Tx for marshalling/unmarshalling into a node tx format.
 //
 // Marshalling usage example:
-//  bb, err := json.Marshal(tx.NodeJSON())
+//
+//	bb, err := json.Marshal(tx.NodeJSON())
 //
 // Unmarshalling usage example:
-//  tx := bt.NewTx()
-//  if err := json.Unmarshal(bb, tx.NodeJSON()); err != nil {}
+//
+//	tx := bt.NewTx()
+//	if err := json.Unmarshal(bb, tx.NodeJSON()); err != nil {}
 func (tx *Tx) NodeJSON() interface{} {
 	return &nodeTxWrapper{Tx: tx}
 }
 
 // NodeJSON returns a wrapped bt.Txs for marshalling/unmarshalling into a node tx format.
 //
 // Marshalling usage example:
-//  bb, err := json.Marshal(txs.NodeJSON())
+//
+//	bb, err := json.Marshal(txs.NodeJSON())
 //
 // Unmarshalling usage example:
-//  var txs bt.Txs
-//  if err := json.Unmarshal(bb, txs.NodeJSON()); err != nil {}
+//
+//	var txs bt.Txs
+//	if err := json.Unmarshal(bb, txs.NodeJSON()); err != nil {}
 func (tt *Txs) NodeJSON() interface{} {
 	return (*nodeTxsWrapper)(tt)
 }
 
-func (tx *Tx) toBytesHelper(index int, lockingScript []byte) []byte {
+func (tx *Tx) toBytesHelper(index int, lockingScript []byte, extended bool) []byte {
```

Comment:
Could extended become a property of Tx?

Reply:
Or an array / map[string]bool of supported protocols the current tx supports or implements?

Then you could do if `tx.protocols[BIPXYZ] { ... }`

Reply:
We did consider this, however, the extended state is not a part of the Tx itself: it is simply a different form of serialisation.   There is no such thing as an extended Tx.